libqtile/bar.py

```diff
@@ -242,9 +243,11 @@ def _configure(self, qtile, screen):
                 if success:
                     qtile.register_widget(i)
 
-            self._remove_crashed_widgets()
-
+        self._remove_crashed_widgets()
+        self.window.unhide()
```

Comment:
Why do we still need this?

Reply:
We only need it with a newly created window, so I've just pushed it up to right after it is made.